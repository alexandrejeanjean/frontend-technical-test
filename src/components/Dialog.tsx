import { Message } from "../types/message";
import { getLoggedUserId } from "../utils/getLoggedUserId";

const Dialog = ({ body, authorId, ...props }: Partial<Message>) => {
  const loggedUserId = getLoggedUserId();
  if (authorId === loggedUserId) {
    return (
      <>
        <span className="text-xs font-medium text-gray-600">{authorId}</span>
        <div className="flex justify-end max-w-full mb-2 ">
          <p className="max-w-sm px-3 py-2 text-blue-900 bg-blue-100 rounded-t-lg rounded-bl-lg ">{body}</p>
        </div>
      </>
    );
  }
  return (
    <>
      <span className="text-xs font-medium text-gray-600">{authorId}</span>
      <div className="max-w-full mb-2 ">
        <p className="max-w-sm px-3 py-2 text-blue-900 bg-gray-100 rounded-t-lg rounded-br-lg">{body}</p>
      </div>
    </>
  );
}

export default Dialog;
