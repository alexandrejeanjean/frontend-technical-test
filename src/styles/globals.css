@tailwind base;
@tailwind components;
@tailwind utilities;

.arrow {
  @apply block p-1 border-t-0 border-b-2 border-l-0 border-r-2 border-gray-500 border-solid;
}

.right {
  transform: rotate(-45deg);
  -webkit-transform: rotate(-45deg);
  @apply border-indigo-600;
}

.main-container-height {
  height: calc(100vh - 132px);
}

.list-height {
  height: calc(100% - 50px);
}