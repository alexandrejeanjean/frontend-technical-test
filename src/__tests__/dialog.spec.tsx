import { expect, jest } from '@jest/globals';
import { mount } from 'enzyme';
import Dialog from "../components/Dialog";

const refetchMessages = jest.fn();

const mockConversation = {
    "id": 1,
    "recipientId": 2,
    "recipientNickname": "Jeremie",
    "senderId": 1,
    "senderNickname": "Thibaut",
    "lastMessageTimestamp": 1625637849
};

const mockMessage = {
    "id": 1,
    "conversationId": 1,
    "timestamp": 1625637849,
    "authorId": 1,
    "body": "Bonjour c'est le premier message de la première conversation"
};

/** @test {Dialog Component} */
describe('Dialog Component', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = mount(<Dialog message={mockMessage} conversation={mockConversation} refetchMessages={refetchMessages} />);
    });

    it('should render without crashing', () => {
        mount(<Dialog message={mockMessage} conversation={mockConversation} refetchMessages={refetchMessages} />);
        expect(wrapper.find('p')).toHaveLength(1);
    });

    it('should display the good value', () => {

        expect((wrapper).prop('message')).toEqual({
            "id": 1,
            "conversationId": 1,
            "timestamp": 1625637849,
            "authorId": 1,
            "body": "Bonjour c'est le premier message de la première conversation"
        });
    });

    it('match snapshot', () => {
        expect(wrapper).toMatchSnapshot()
    })

});
