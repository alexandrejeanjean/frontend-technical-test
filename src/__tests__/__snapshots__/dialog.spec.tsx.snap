// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog Component match snapshot 1`] = `ReactWrapper {}`;
